airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -20,14 +20,22 @@
 import json
 import warnings
 from time import sleep
-from typing import Any, Callable
+from typing import TYPE_CHECKING, Any, Callable
 
 from botocore.exceptions import ClientError
 
 from airflow.compat.functools import cached_property
 from airflow.exceptions import AirflowException, AirflowNotFoundException
 from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
 
+if TYPE_CHECKING:
+    from mypy_boto3_emr.literals import ClusterStateType
+    from mypy_boto3_emr.type_defs import (
+        ListClustersOutputTypeDef,
+        RunJobFlowInputRequestTypeDef,
+        RunJobFlowOutputTypeDef,
+    )
+
 
 class EmrHook(AwsBaseHook):
```

Comment:
Also suggestion change base class for EmrHook to the same way as it done for RdsHook:
https://github.com/apache/airflow/blob/7efdeed5eccbf5cb709af40c8c66757e59c957ed/airflow/providers/amazon/aws/hooks/rds.py#L23-L29

It would allow get type annotation to `EmrHook.conn` property and `EmrHook.get_conn()` method

Reply:
Done!